File: java/src/Problem9.java
/**
 * Created by adam on 2/1/14.
 */
public class Problem9 implements Euler{
    final int sum = 1000;
    @Override
    public String answer() {
        for(int c = 3; c < sum; c++) {
            for(int b = 2; b < c; b++) {
                for(int a = 1; a < b; a++) {
                    if (((int)(Math.pow(a, 2) + Math.pow(b, 2)) == (int)Math.pow(c, 2)) && (a + b + c == sum)) {
                        return Integer.toString(a*b*c);
                    }
                }
            }
        }
        return "No answer";
    }
}

File: dart/lib/src/Problem9.dart
import "Euler.dart";
import "dart:math";

class Problem9 implements Euler {
  String answer() {
    final int sum = 1000;

    for (int c = 3; c < sum; c++) {
      for (int b = 2; b < c; b++) {
        for (int a = 1; a < b; a++) {
          if (pow(a, 2) + pow(b, 2) == pow(c, 2) && (a + b + c == sum)) {
            return (a * b * c).toString();
          }
        }
      }
    }

    return null;
  }
}
